hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/management/ElasticsearchLogRepository.java

```diff
@@ -0,0 +1,122 @@
+package pl.allegro.tech.hermes.tracker.elasticsearch.management;
+
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.sort.SortOrder;
+import pl.allegro.tech.hermes.api.MessageTrace;
+import pl.allegro.tech.hermes.api.PublishedMessageTrace;
+import pl.allegro.tech.hermes.api.PublishedMessageTraceStatus;
+import pl.allegro.tech.hermes.api.SentMessageTrace;
+import pl.allegro.tech.hermes.api.SentMessageTraceStatus;
+import pl.allegro.tech.hermes.tracker.elasticsearch.ElasticsearchRepositoryException;
+import pl.allegro.tech.hermes.tracker.elasticsearch.LogSchemaAware;
+import pl.allegro.tech.hermes.tracker.management.LogRepository;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+
+public class ElasticsearchLogRepository implements LogRepository, LogSchemaAware {
+
+    private static final int LIMIT = 1000;
```

Comment:
this could be a configurable option mostly for performance reasons because from what I remember how Lucene works less data we fetch faster the queries are